mlflow/pyfunc/scoring_server/__init__.py

```diff
@@ -323,6 +323,15 @@ def invocations(data, content_type, model, input_schema):
             _log_warning_if_params_not_in_predict_signature(_logger, params)
             raw_predictions = model.predict(data)
     except MlflowException as e:
+        if "Failed to enforce schema" in e.message:
+            _logger.warning(
+                "If using `instances` as input key, we internally convert "
+                "the data type from `records` (List[Dict]) type to "
+                "`list` (Dict[str, List]) type if the data is a pandas "
```

Comment:
If `records` is `List[Dict[str, T]]`, what would `list` look like?

Reply:
It's Dict[str, List[T]]. It's appending values with the same key to construct a list.